woocommerce/woocommerce-ios
# [Payments Menu] Background onboarding support

Closes: #11104


## Description


As part of the rewrite of the Payments Menu in SwiftUI, this PR adds the background onboarding support to the new menu.

This includes:
- A background check which runs when the Payments Menu is shown, if we don't have `.completed` in the cache
- A spinner which shows while a check is ongoing (moved)
- A notice which displays when onboarding has steps to complete or other issues
- The ability to open the onboarding screens from the notice
- Disabling/enabling the Manage Card Readers link based on onboarding status

Additionally, the spinner is relocated to the navigation bar, instead of being in the background of the list. This is because on many phones, since adding the Tap to Pay rows, we no longer have space in the background of the list where users can see the spinner.

### Known issues
- The notice doesn't animate in and out #11129 
- Learn More links on the onboarding screens don't work #11131 

## Testing instructions


1. Use a store with both Stripe and WooPayments installed, but deactivated
2. Delete the app, then launch it from Xcode.
3. Log in to a store other than the one you set up above
4. Navigate to `Menu > Settings > Experimental Features`
5. Toggle `SwiftUI Payments Menu`
6. Switch to your test store from step 1
7. Navigate to `Menu > Payments`
8. Observe that the spinner is shown as onboarding takes place, the `Manage Card Reader` row is disabled.
9. Observe that after a couple of seconds, the Onboarding notice is shown.
10. Tap it, observe that the onboarding screen is shown inviting you to activate WooPayments.
11. In WP-admin, activate Stripe
12. In the app, tap to activate WooPayments
13. Observe that the plugin selector is shown – select a gateway and tap the CTA
14. Observe that the notice is removed, `Manage Card Reader` is enabled, and a `Payment Provider` row is added
15. Tap the Payment Provider row and observe that you can change the selected gateway.
16. Observe that onboarding tracks events have been logged as expected.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/056e6ce1-694e-47dc-82e2-b22cbe49dd37


---
